src/app/modules/project/embed-button/embed-button.component.ts

```diff
@@ -0,0 +1,49 @@
+import { EmbedService } from './../../../services/embed.service';
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { HttpErrorResponse } from '@angular/common/http';
+import { EmbeddedProject } from 'src/app/models/domain/embedded-project';
+import { environment } from 'src/environments/environment';
+
+
+@Component({
+  selector: 'app-embed-button',
+  templateUrl: './embed-button.component.html',
+  styleUrls: ['./embed-button.component.scss']
+})
+/**
+ * Button that generates an embedded project and creates an iframe example link
+ */
+export class EmbedButtonComponent implements OnInit {
+  EmbeddedProject: EmbeddedProject;
+  frontendUrl = environment.frontendUrl;
+
+  private ProjectId: number;
+
+  constructor(private activedRoute: ActivatedRoute, private EmbedService: EmbedService) { }
+
+
+  ngOnInit() {
+    const routeId = this.activedRoute.snapshot.paramMap.get("id");
+    if (!routeId) {
+      return;
+    }
+    const id = Number(routeId);
+    if (id < 1) {
+      return;
+    }
+    this.ProjectId = id;
+  }
+
+  generateEmbed(){
+    this.EmbedService.post({ProjectId:this.ProjectId})
+      .subscribe(
+        (result) => {
+          this.EmbeddedProject = result;
+        },
+        (error: HttpErrorResponse) => {
+          //handle error
```

Comment:
Do we need a specific error to be thrown in this case? Or will the upcoming global error handler be sufficient enough?

Reply:
The global handler should be sufficient, this is why i did not implement something custom.

Reply:
In that case you can drop the error clause